StorageUtility/DeviceInterface.h

```diff
@@ -0,0 +1,50 @@
+/*
+<License>
+Copyright 2015 Virtium Technology
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http ://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+</License>
+*/
+#ifndef __DeviceInterface_h__
+#define __DeviceInterface_h__
+#pragma once
+
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "BasicTypes.h"
+#include "vtStorPlatformDefines.h"
+
+namespace vtStor
+{
+
+class cDeviceInterface
+{
+public:
+    cDeviceInterface();
+    virtual ~cDeviceInterface();
+
+public:
+    std::map<U32, void*> Data();
```

Comment:
Does "data" needs to be modified or read-only? Be very very careful about passing "containers" around. Making copies is not cheap.